File: Back/CourseManagementApp/Model/Training.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace CourseManagementApp.Model
{
    public class Training : IEntity<Guid>
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }
        [Required]
        [MaxLength(1000)]
        public string Description { get; set; }
        public virtual ICollection<Course> Courses { get; set; }
    }
}

File: Front/src/app/shared/model/training.ts
import { Course } from "./course";

export class Training  {
    name: string = "";
    description: string = "";
    courses: Course[] = [];

    public constructor(obj?: any) {
        if (obj) {
          this.name = obj.name;
          this.description = obj.description;
          this.courses = obj.courses;
        }
      }
    
}